### PR TITLE
SDK-1401. Generating thumbnails and previews for webP images (iOS)

### PR DESCRIPTION
**Pull request by jnavarrom**

### CODE REVIEW

**Review by lhr000lhrmega — APPROVED** [review]

**Comment by lhr000lhrmega** [comment]
I was wondering is it possible that not hard code these supported types.
Based on my test, Image I/O(on  iOS 11) does not crash or show any side effect, even we added `webp` here.

File: src/gfx/GfxProcCG.mm [hunk 1]
```diff
@@ -62,7 +62,7 @@
     if ([[NSBundle.mainBundle objectForInfoDictionaryKey:@"CFBundleExecutable"] isEqualToString:@"MEGAFiles"]) {
         return "";
     }
-    return ".bmp.cr2.crw.cur.dng.gif.heic.ico.j2c.jp2.jpf.jpeg.jpg.nef.orf.pbm.pdf.pgm.png.pnm.ppm.psd.raf.rw2.rwl.tga.tif.tiff.3g2.3gp.avi.m4v.mov.mp4.mqv.qt.";
+    return ".bmp.cr2.crw.cur.dng.gif.heic.ico.j2c.jp2.jpf.jpeg.jpg.nef.orf.pbm.pdf.pgm.png.pnm.ppm.psd.raf.rw2.rwl.tga.tif.tiff.3g2.3gp.avi.m4v.mov.mp4.mqv.qt.webp.";
```
Comment (simonwng) [inline src/gfx/GfxProcCG.mm:65]: since we have error handling in Image I/O, shall we remove this format list?

**Review by simonwng — APPROVED** [review]

File: src/gfx/GfxProcCG.mm (hunk 1 above)
Comment (jnavarrom) [reply src/gfx/GfxProcCG.mm:65]: We can't remove this line. 
We have to implement the following virtual method:

```
// list of supported extensions (NULL if no pre-filtering is needed)
virtual const char* supportedformats();
```

To remove it, we would need changes in the SDK

Comment (simonwng) [reply src/gfx/GfxProcCG.mm:65]: It has a comment:
```
// list of supported extensions (NULL if no pre-filtering is needed)
```
It seems if we return NULL, there will be no filtering?

Comment (jnavarrom) [reply src/gfx/GfxProcCG.mm:65]: This is how the others implementations of the gfx layer implement `supportedFormat()`:

https://github.com/meganz/sdk/blob/develop/src/gfx/freeimage.cpp#L383
https://github.com/meganz/sdk/blob/develop/src/gfx/qt.cpp#L1425

Comment (simonwng) [reply src/gfx/GfxProcCG.mm:65]: https://jira.developers.mega.co.nz/browse/SDK-1405 For further investigation of removing the format list

**Pull request merged**